app/models/heart_rate_data.rb

```diff
@@ -0,0 +1,30 @@
+class HeartRateData < ApplicationRecord
+
+  belongs_to :user, dependent: :destroy
+
+  def update_heart_rate_data!
+    update!(data: heart_rate_data_for_the_day)
+  end
+
+  private
+
+  def heart_rate_data_for_the_day
```

Comment:
Please update this method to take an optional `date` parameter. Make sure it conforms to the 'YYYY-mm-dd' format required by the API